File: insight_desktop/engine/visual_context.py
"""Active photo context for the current conversation."""

from __future__ import annotations

from dataclasses import dataclass


@dataclass
class VisualContext:
    image_path: str
    caption: str

    def prompt_block(self) -> str:
        return (
            "What the user is showing you (from their attached photo — "
            "keep this in mind for follow-up questions until they attach a new photo):\n"
            f"{self.caption}"
        )

File: insight_ios/Packages/InsightCore/Sources/InsightCore/VisualContext.swift
import Foundation

/// Active photo context for the current conversation.
public struct VisualContext: Sendable, Equatable {
    public let imagePath: String
    public let caption: String

    public init(imagePath: String, caption: String) {
        self.imagePath = imagePath
        self.caption = caption
    }

    public func promptBlock() -> String {
        """
        What the user is showing you (from their attached photo — keep this in mind for follow-up questions until they attach a new photo):
        \(caption)
        """
    }
}
